### PR TITLE
Add single task execution create request validation

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Because single task executions don't use launch plans, some parameters that are optional overrides for conventional ExecutionCreateRequests are actually mandatory.

Note: the reference entity used to launch an execution is confusingly named LaunchPlan but accepts identifiers of task resouce_type.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/567

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/execution_validator.go [hunk 1]
```diff
@@ -157,3 +162,12 @@ func ValidateWorkflowExecutionIdentifier(identifier *core.WorkflowExecutionIdent
 	}
 	return nil
 }
+
+// Because single task executions don't use launch plans, some parameters that are optional overrides for conventional
+// ExecutionCreateRequests are actually mandatory.
+func validateLaunchSingleTaskExecutionReq(request admin.ExecutionCreateRequest) error {
+	if request.Spec.AuthRole == nil {
```
Comment (EngHabu) [inline pkg/manager/impl/validation/execution_validator.go:169]: Should this check inside the AuthRole as well? like `AuthRole.Method.....`

Comment (katrogan) [reply pkg/manager/impl/validation/execution_validator.go]: excellent point, done

**Review by EngHabu — APPROVED** [review]
